### PR TITLE
Balancer Liquidity Update

### PR DESCRIPTION
**Pull request by viniabussafi**
# Context
This PR comprehends multiple changes across different files with the intention of shielding our liquidity spells from outlier data on token prices. 
What motivated these changes was a discrepancy on the price for OATH, for the [OATH/WETH 80/20 pool on Optimism](https://op.beets.fi/pool/0xd20f6f1d8a675cdca155cb07b5dc9042c467153f0002000000000000000000bc). Balancer frontend had a TVL of approx. 800k while dune displayed a TVL of more than 12 million on 2023-11-30, as shown on this [query](https://dune.com/queries/3259661). With the changes on this PR, it seems fixed, per this [query](https://dune.com/queries/3260199).
From my understanding, this comes from a difference in prices shown in [coinpaprika](https://coinpaprika.com/coin/oath-oath/) and other providers, like [coinmarketcap](https://coinmarketcap.com/currencies/oath/) on that date. I think this is something that the dune team could communicate to coinpaprika.
To avoid that in the future, I have created a whitelist of pricing assets, due to their higher volume and trustability of data. With this whitelist, we can calculate the liquidity on weighted pools based on the price of these pricing assets and their respective weight.

Also, I have created the `protocol_liquidity_eth` and `pool_liquidity_eth` columns, which will be extremely useful now that ethereum prices have been volatile and this can skew our usd data.

# Main changes made to each file

### balancer_token_whitelist
- Created file with selected tokens, their addresses and respective chains

### balancer_v2_{{chain}}_pools_tokens_weights:
- Added a blockchain column
- removed post_hook to clean the data explorer

### balancer_{{chain}}_pools_tokens_weights:
- Added a blockchain column

### balancer_pools_tokens_weights:
- Created multichain spell, which will be useful with balancer_v3 coming next year

### balancer_v2_{{chain}}_liquidity
- Created macro
- Added `eth_prices` CTE, to be used on final select statement
- Created `weighted_pool_liquidity_estimates` and `weighted_pool_liquidity_estimates_2` CTEs, to calculate the liquidity of pricing assets on weighted pools on the first CTE, while the second CTE avoids double count on pools with multiple pricing assets.
- Added coalesce operands to `pool_liquidity_usd` and `protocol_liquidity_usd`, to also account for the cases where a pool is not weighted or doesn't have a pricing asset
- Added `protocol_liquidity_eth` and `pool_liquidity_eth`

### balancer_liquidity
- Added `protocol_liquidity_eth` and `pool_liquidity_eth`

@thetroyharris @mendesfabio, would appreciate your review before passing it along to the dune team

### CODE REVIEW

File: models/balancer/avalanche_c/balancer_v2_avalanche_c_liquidity.sql [hunk 1]
```diff
@@ -1,236 +1,21 @@
...
+        schema='balancer_v2_' + blockchain,
         alias = 'liquidity',
-        
         materialized = 'table',
         file_format = 'delta',
-        post_hook='{{ expose_spells(\'["avalanche_c"]\',
-                                    "project",
-                                    "balancer_v2",
-                                    \'["stefenon", "viniabussafi"]\') }}'
+        post_hook='{{ expose_spells(\'["{{blockchain}}"]\',
```
Comment (thetroyharris) [inline models/balancer/avalanche_c/balancer_v2_avalanche_c_liquidity.sql:10]: In the arb v2 liq spell you have `expose_spells(\'["arbitrum"]\',` 
Just noting here so can change one of them to match the other.

**Comment by viniabussafi** [comment]
hey, @jeff-dude! ready for review when you can :)

**Comment by jeff-dude** [comment]
hi @viniabussafi -- really nice cleanup work here!

i have one piece of feedback, curious your thoughts on implementing in this lineage of spells.

in our latest design patterns, like dex/nft beta sector spells, we've been removing all "metadata" joins in chain-specific spells, and waiting to add until the final step post-union of chains to build the cross-chain spell. when i say metadata, think prices, dex prices, erc20 tokens, anything not specific to balancer decoded/raw tables. the main goal here is to ease the orchestration in prod and centralize the metadata additions, especially if that final cross-chain is the only one meant to be queried on the app.

we've gotten feedback that chain-specific spells are still nice to query as well, so we took the approach of "staging" models throughout the lineage to build the final cross-chain spell with above comments in mind. then at the end, build a simple view spell on top of that cross-chain spell for each chain, as needed, for easy queries.

let me know if you think that'll make sense here 🙏

**Comment by jeff-dude** [comment]
> Hey @jeff-dude! Thanks for your review!
> 
> So, this is something me and @mendesfabio have been looking into for the past few weeks and would definitely like to implement at some point in the near-future. If it works for you, it'd be great if we could have this PR merged as-is, just so we can handle some of the issues we have at the moment and as soon as I have some bandwidth I can start working on the changes you suggested.
> 
> Thank you very much once again

that works for me. looking forward to seeing the next round of enhancements 🤝

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**